processing/src/main/java/org/apache/druid/java/util/common/Intervals.java

```diff
@@ -53,6 +57,74 @@ public static Interval of(String format, Object... formatArgs)
     return of(StringUtils.format(format, formatArgs));
   }
 
+  /**
+   * A performance-optimized method for parsing a Joda-Time {@link Interval} from a string.
+   * This method is significantly faster than the standard {@link Intervals#of(String)} for the following
+   * group of offsets:
+   * <ol>
+   *   <li>"2022-01-01T00:00:00.000Z/2022-01-02T00:00:00.000Z"</li>
+   *   <li>"2022-01-01T00:00:00.000+05:30/2022-01-01T01:00:00.000+05:30"</li>
+   *   <li>"2022-01-01T00:00:00.000+0530/2022-01-01T01:00:00.000+0530"</li>
+   * </ol>
+   * <p>
+   * If the input string does not match the format, it will fall back to the more flexible but
+   * slower {@link Intervals#of(String)} parser. If you are dealing with any Intervals format examples below,
+   * consider using {@link Intervals#of(String)} instead:
+   * <ol>
+   *   <li>"2022-01-01T00:00:00Z/2022-01-02T00:00:00Z" (without millis)</li>
+   *   <li>"2022-01-01/2022-01-02" (Date only)</li>
+   *   <li>"2022-01-01T12:00:00.000Z/PT6H" (Periods in start / end)</li>
+   * </ol>
+   *
+   * Currently, this method is only used in {@link org.apache.druid.timeline.SegmentId}.
+   * Should you find additional places where the Interval format is guaranteed to be compatible with this method,
+   * feel free to open a PR.
```

Comment:
Nit: We can leave out these lines.

Reply:
Since this method is an optimization and used by the DataSegment only, my opinion is to move this method from the Interval class to the DataSegment.

Reply:
Hi @FrankChen021 

While this method is only used in `DataSegment` now, should another place find this deserialization useful, we may need to move the method back to `Intervals`.

Furthermore, I feel that this we should place a method that deals with interval parsing in `Intervals`. What do you think?